### PR TITLE
fix(settings): Figma feature-gate popup for tier-locked Custom RPC (#4973)

### LINKED ISSUE #4973: Tier-gate popup: wrong description, should follow Figma as a reusable feature-gate component
## Problem
The locked-feature / tier-gate popup uses the **wrong description/layout** and doesn't match the Figma design ([node 77059-110340](https://www.figma.com/design/puB2fsVpPrBx3Sup7gaa3v/Vultisig-App?node-id=77059-110340)).

Tapping a tier-locked feature currently shows a generic **tier-discount** sheet ("Unlock <Tier> — …receive N bps trading fee discount…"). The Figma is a **feature-gate** popup: the gated **feature** name + description, a "Requires **$VULT <Tier> or above** — Hold at least N VULT" card, the user's **balance**, and a **Get $VULT** CTA.

> **Note:** the required tier stays whatever the gate currently uses (e.g. Silver) — the "Gold Tier" shown in the Figma frame is just the reference example. Keep the actual tier; the component just swaps it in.

## Ask
Rebuild the tier-gate popup to the Figma design as **one reusable component**, parameterized by the gated feature (title + description) + required tier (+ VULT threshold) + balance — swap the tier/feature per gate.

## Where (Android)
- Strings `vault_tier_*_description_part*`, `vault_tier_unlock` in `app/src/main/res/values/strings.xml` (current tier-discount framing).
- Discount-tier feature: `app/src/main/java/com/vultisig/wallet/ui/screens/settings/DiscountTiersScreen.kt` + `DiscountTiersViewModel.kt`.

Parallel: iOS https://github.com/vultisig/vultisig-ios/issues/4607 · Windows https://github.com/vultisig/vultisig-windows/issues/4156

### PR DESCRIPTION
**Pull request by aminsato**
## Summary

Closes #4973.

Tapping the tier-locked **Custom RPC** entry showed the generic `$VULT` **discount** sheet (_"Unlock <Tier> — receive N bps trading fee discount…"_) instead of a **feature-gate** that explains the gated feature and its tier requirement.

The Figma feature-gate design ([node 77059-110340](https://www.figma.com/design/puB2fsVpPrBx3Sup7gaa3v/Vultisig-App?node-id=77059-110340)) was **already implemented** for the Advanced Swap gate (`SwapAdvancedSettingsLockedSheet`, #4969). Rather than add a second near-identical component, this **generalizes it into one reusable `FeatureGateBottomSheet`** — parameterized by feature (icon/title/description), required tier, and the vault's balance/threshold — and routes **both** the Advanced Swap upsell and the Custom RPC upsell through it. The tier badge, name, and CTA accent all derive from the required tier (`getStyleByTier`), so it works for any tier.

## Changes
- **New `FeatureGateBottomSheet`** — single source of truth for the locked-feature design.
- **`SwapAdvancedSettingsLockedSheet`** reduced to a thin wrapper over it (no behavior change).
- **Custom RPC gate** (`SettingsScreen` + `SettingsViewModel`) now renders it with the vault's live `$VULT` balance and the Silver threshold; the required tier stays **Silver** as before.
- **Strings** — added generic `feature_gate_*` + `custom_rpc_gate_description`; removed the now-redundant `swap_advanced_locked_*` label strings. Updated all 10 locales.
- **PreviewActivity** — added a `custom_rpc_gate` case for screenshots.

## Screenshot

Custom RPC feature-gate (below Silver tier), on device:

custom-rpc-gate

## Test plan
- [x] `:app:compileDebugKotlin` builds (incl. debug source set)
- [x] ktfmt applied; locale placeholders consistent across all 10 files
- [x] Device-verified: Settings → Custom RPC (below Silver) shows the new feature-gate popup
- [ ] Advanced Swap → Advanced Settings (below Silver) still shows its upsell unchanged

## Summary by CodeRabbit

* **New Features**
  * Added a Custom RPC upsell flow that gates access using your VULT balance vs. a tier threshold, showing current balance and required hold amount.
  * Introduced a reusable “feature gate” bottom sheet for tier-locked features with dismiss/CTA actions and below-threshold warning.
* **Refactor**
  * Updated the swap advanced “locked” bottom sheet to use the shared feature gate component.
* **Localization**
  * Added/updated localized copy for the new Custom RPC and feature gate messaging across multiple languages.
* **Tests**
  * Expanded Custom RPC view-model tests and added balance formatting coverage.

### CODE REVIEW

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        // Tier-accent "Get $VULT" footer joined to the card's bottom edge (single rounded
+        // container). Darker accent at the top fading to the bright tier accent at the bottom.
+        Box(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .height(52.dp)
+                    .background(
+                        Brush.verticalGradient(
+                            listOf(
+                                lerp(tierStyle.accentColor, Color.Black, 0.4f),
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt:204]: **Visual change to the already-shipped Advanced Swap footer.** The old sheet used explicit Silver constants (`top 0xFF7D8B9E` → `bottom 0xFFC9D6E8`). This derives the gradient as `lerp(accentColor, Black, 0.4f)` → `accentColor`. For Silver the bottom stop is identical (`0xFFC9D6E8`) but the top stop shifts from `0xFF7D8B9E` to ≈`0xFF78808B`. It now matches the `DiscountTiersScreen` pattern (arguably more consistent), but "Advanced Swap unchanged" is no longer strictly true — please verify the Advanced Swap footer against Figma node 77059-110340 on device. (The PR test plan leaves that box unchecked.)

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+ * @param requiredTier tier required to unlock the feature; drives the badge, name, and CTA accent.
+ * @param balanceText the vault's current $VULT balance, pre-formatted with its unit (e.g. "2,340
+ *   VULT").
+ * @param thresholdText the required $VULT holdings, pre-formatted with its unit (e.g. "3,000
+ *   VULT").
+ * @param isBelowThreshold when true the balance is rendered in the warning accent.
+ */
+@Composable
+internal fun FeatureGateBottomSheet(
+    featureIcon: Int,
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt]: Minor: annotate as `@DrawableRes featureIcon: Int` so lint can catch a non-drawable passed here. Every other param is a real type.

File: app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt [hunk 3]
```diff
@@ -359,7 +366,16 @@ constructor(
                     if (isSilver) {
                         navigator.route(Route.CustomRpcList(vaultId))
                     } else {
-                        state.update { it.copy(showCustomRpcUpsell = true) }
+                        val balanceRaw =
+                            runCatching { getDiscountBps.getVultBalance(vaultId) }.getOrNull()
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt]: Minor perf: the VULT balance is now fetched twice per tap — once inside `getDiscountBps.hasReachedSilverTier(vaultId)` and again here. Consider fetching once and deriving `isSilver` locally:

```kotlin
val balance = runCatching { getDiscountBps.getVultBalance(vaultId) }.getOrNull() ?: BigInteger.ZERO
if (balance >= SILVER_TIER_THRESHOLD) navigator.route(...) else state.update { ... }
```

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/SettingsScreen.kt [hunk 4]
```diff
@@ -132,10 +132,16 @@ private fun SettingsScreen(
...
-                onContinue = onUnlockCustomRpcTier,
-                onDismissRequest = onDismissCustomRpcUpsell,
+            FeatureGateBottomSheet(
+                featureIcon = R.drawable.settings_globe,
+                featureTitle = stringResource(R.string.custom_rpc_title),
+                featureDescription = stringResource(R.string.custom_rpc_gate_description),
+                requiredTier = TierType.SILVER,
+                balanceText = state.customRpcVultBalance,
+                thresholdText = state.customRpcVultThreshold,
+                isBelowThreshold = true,
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/ui/screens/settings/SettingsScreen.kt]: `isBelowThreshold = true` is hardcoded. Correct today (the sheet only renders in the not-Silver branch), but it bakes a precondition into the UI layer. If you adopt the single-fetch suggestion in the ViewModel you'll already have the balance to pass a real `balance < threshold` — removes a latent footgun for future reuse of this gate.

File: app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt [hunk 5]
```diff
@@ -554,3 +570,11 @@ constructor(
         viewModelScope.launch { navigator.route(Route.DiscountTiers(vaultId)) }
     }
 }
+
+/** Formats a raw VULT balance (18 decimals) as a grouped, whole-token string e.g. "2,340 VULT". */
+private fun formatVultBalance(raw: BigInteger): String {
```
Comment (johnnyluo) [inline app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt:576]: `formatVultBalance` is a pure function with locale/decimal edge cases (grouping, rounding to 0 fraction digits). A small JVM unit test would be cheap insurance — not blocking since it's display-only. (Verified the math is correct: `VULT.decimal = 18`, `SILVER_TIER_THRESHOLD = 3000·10^18` → `"3,000 VULT"`.)

**Review by johnnyluo — COMMENTED** [review]
A few inline notes from the review. The refactor is clean — main thing I'd want before merge is on-device confirmation of the Advanced Swap footer (its gradient derivation changed). Other comments are non-blocking.

File: app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt [hunk 6]
```diff
@@ -359,7 +366,16 @@ constructor(
...
+                        val balanceRaw =
+                            runCatching { getDiscountBps.getVultBalance(vaultId) }.getOrNull()
+                                ?: BigInteger.ZERO
+                        state.update {
+                            it.copy(
+                                showCustomRpcUpsell = true,
+                                customRpcVultBalance = formatVultBalance(balanceRaw),
+                                customRpcVultThreshold = formatVultBalance(SILVER_TIER_THRESHOLD),
+                            )
+                        }
```
[the remark this answers is not in the document]
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt]: Done in 958ee63 — the Custom RPC handler now uses `viewModelScope.safeLaunch { ... }` for the `getVultBalance` repository I/O.

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt (hunk 1 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt:204]: Good catch — flagged for on-device verification. The shared component derives the footer gradient as `lerp(accentColor, Black, 0.4f) → accentColor`, so for Silver the bottom stop is unchanged (`0xFFC9D6E8`) but the top stop shifts `0xFF7D8B9E → ≈0xFF78808B` to match the `DiscountTiersScreen` pattern. I still need to verify the Advanced Swap footer against Figma node 77059-110340 on device before merge — leaving that test-plan box unchecked until then.

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt (hunk 2 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt]: Done in 958ee63 — `featureIcon` is now `@DrawableRes featureIcon: Int`.

File: app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt (hunk 3 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt]: Done in 958ee63 — now fetches the balance once and derives the tier locally (`balanceRaw >= SILVER_TIER_THRESHOLD`), dropping the redundant `hasReachedSilverTier` call. Wrapped in `safeLaunch`.

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/SettingsScreen.kt (hunk 4 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/screens/settings/SettingsScreen.kt]: Done in 958ee63 — added `customRpcIsBelowThreshold` to state, computed in the VM as `balanceRaw < SILVER_TIER_THRESHOLD` (now possible thanks to the single-fetch), and the screen passes `state.customRpcIsBelowThreshold` instead of the hardcoded `true`.

File: app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt (hunk 5 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/models/settings/SettingsViewModel.kt:576]: Done in 958ee63 — added a locale-pinned test (`Locale.US`) asserting `2,340 VULT` / `3,000 VULT`. Also note the tier thresholds are now pure `BigInteger` math instead of `CoinType.ETHEREUM.toUnit(...)`; the native call was crashing the upsell unit tests during class init, which was the CI build failure.

**Review by johnnyluo — APPROVED** [review]
LGTM

File: app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt (hunk 1 above)
Comment (johnnyluo) [reply app/src/main/java/com/vultisig/wallet/ui/screens/settings/bottomsheets/FeatureGateBottomSheet.kt:204]: Accepting as by-design: unifying the footer gradient on the shared `lerp(accentColor, Black, 0.4f) → accentColor` derivation (matching the `DiscountTiersScreen` tier pattern) is intentional. For Silver the bottom stop is unchanged; the ~`0xFF7D8B9E → 0xFF78808B` top-stop shift is an accepted, negligible visual delta. Resolving.
